sonar-plugin-api/src/main/java/org/sonar/api/server/ws/internal/ValidatingRequest.java

```diff
@@ -139,6 +145,9 @@ private String readParamOrDefaultValue(String key, @Nullable WebService.Param de
   @CheckForNull
   protected abstract InputStream readInputStreamParam(String key);
 
+  @CheckForNull
+  protected abstract Part readPart(String key);
```

Comment:
Good question, in fact it's not always possible to have a Part (see in LocalRequestAdapter : https://github.com/SonarSource/sonarqube/pull/1071/commits/68dcc5a1143353549864c3c9b78160e79f32c742#diff-d0a2957377356e98dd5324c001af871cR52).
We should handle it later, but I don't how to do that for the moment...